flake.nix

```diff
@@ -0,0 +1,189 @@
+{
+  description = "Static analyser of semantic differences in large C projects";
+
+  inputs = { nixpkgs.url = "github:NixOS/nixpkgs/release-23.05"; };
+
+  outputs = { self, nixpkgs, ... }:
+    let
+      system = "x86_64-linux";
+      pkgs = import nixpkgs { inherit system; };
+
+      llvmVersionMin = 9;
+      llvmVersionMax = 16;
+      llvmVersions = pkgs.lib.lists.range llvmVersionMin llvmVersionMax;
+
+      mkDiffkemp =
+        llvmPackages:
+          with pkgs;
+          python3Packages.buildPythonPackage {
+            pname = "diffkemp";
+            version = "0.5.0";
+
+            src = self;
+
+            nativeBuildInputs = with llvmPackages; [
+              cmake
+              gcc
+              libllvm
+              ninja
+              nodejs_20
+            ];
+
+            buildInputs = with llvmPackages; [
+              clangNoLibcxx
+              cscope
+              diffutils
+              gtest
+              gnumake
+            ];
+
+            propagatedBuildInputs = with python3Packages; [
+              cffi
+              pyyaml
+              setuptools
+            ];
```

Comment:
One more thing that occurred to me is that we will have to update this list when we add something to `requierments.txt`. I do not know if it is possible that the nix would take the list of packages from the `requirements.txt` file?

Reply:
https://nix-community.github.io/dream2nix/options/pip.html#opt-pip.requirementsFiles

There is a project that tries to automate it.

Reply:
This is a good point, thanks @DanielKriz for the reference. I wouldn't complicate things at this point, though, as CI should capture this in most cases. But we can certainly revisit this later, there's also a widely-used [poetry2nix](https://github.com/nix-community/poetry2nix) for converting Python packages to Nix but that doesn't work for setuptools-based projects.

Reply:
Thank  @DanielKriz. @viktormalik you are right that CI should capture it, I did not think of that.